examples/locks/locks.bal

```diff
@@ -0,0 +1,45 @@
+import ballerina/io;
+
+// Shared counter variable among multiple workers.
+int counter;
+
+function main (string... args) {
+    foreach i in [1..10] {
+        counterUp();
+    }
+    io:println("final counter value - ", counter);
+}
+function counterUp() {
+    worker w1 {
+        lock {
+            // Lock the shared variable and increment the counter.
+            foreach i in [1..1000] {
+                counter = counter + 1;
```

Comment:
The idea will be more clearer if the lock is inside the loop. Please put it there. Also, the outer loop in the main is not required. You can rename the "counterUp" method to something like "process".